core/src/avm1/activation.rs

```diff
@@ -585,12 +585,20 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
         // ECMA-262 s. 11.6.1
         let a = self.context.avm1.pop();
         let b = self.context.avm1.pop();
-        let result: Value<'_> = if let Value::String(a) = a {
-            AvmString::concat(self.context.gc_context, b.coerce_to_string(self)?, a).into()
-        } else if let Value::String(b) = b {
-            AvmString::concat(self.context.gc_context, b, a.coerce_to_string(self)?).into()
-        } else {
-            (b.coerce_to_f64(self)? + a.coerce_to_f64(self)?).into()
+        let result: Value<'_> = match (a, b) {
+            (Value::String(a), b) => AvmString::concat(
```

Comment:
can we keep a fast path for (string + string)?

Reply:
I think this would get optimized because `to_primitive` etc. simply return `self`, but certainly a good idea to make it explicit.